[Bug] Failed to build BRPC

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.


### Version

master

### What's Wrong?

```shell
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0xae): undefined reference to `google::FlagRegisterer::FlagRegisterer<bool>(char const*, char const*, char const*, bool*, bool*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x12c): undefined reference to `google::FlagRegisterer::FlagRegisterer<bool>(char const*, char const*, char const*, bool*, bool*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x199): undefined reference to `google::FlagRegisterer::FlagRegisterer<bool>(char const*, char const*, char const*, bool*, bool*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x209): undefined reference to `google::FlagRegisterer::FlagRegisterer<bool>(char const*, char const*, char const*, bool*, bool*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x234): undefined reference to `google::FlagRegisterer::FlagRegisterer<int>(char const*, char const*, char const*, int*, int*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x391): undefined reference to `google::FlagRegisterer::FlagRegisterer<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x437): undefined reference to `google::FlagRegisterer::FlagRegisterer<bool>(char const*, char const*, char const*, bool*, bool*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x4a0): undefined reference to `google::FlagRegisterer::FlagRegisterer<int>(char const*, char const*, char const*, int*, int*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x503): undefined reference to `google::FlagRegisterer::FlagRegisterer<int>(char const*, char const*, char const*, int*, int*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x568): undefined reference to `google::FlagRegisterer::FlagRegisterer<int>(char const*, char const*, char const*, int*, int*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x5cd): undefined reference to `google::FlagRegisterer::FlagRegisterer<int>(char const*, char const*, char const*, int*, int*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x727): undefined reference to `google::FlagRegisterer::FlagRegisterer<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x7c5): undefined reference to `google::FlagRegisterer::FlagRegisterer<int>(char const*, char const*, char const*, int*, int*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0x957): undefined reference to `google::FlagRegisterer::FlagRegisterer<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0xae6): undefined reference to `google::FlagRegisterer::FlagRegisterer<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0xb84): undefined reference to `google::FlagRegisterer::FlagRegisterer<int>(char const*, char const*, char const*, int*, int*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0xbf1): undefined reference to `google::FlagRegisterer::FlagRegisterer<bool>(char const*, char const*, char const*, bool*, bool*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0xd42): undefined reference to `google::FlagRegisterer::FlagRegisterer<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0xed3): undefined reference to `google::FlagRegisterer::FlagRegisterer<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-logging.o):(.text.startup+0xf71): undefined reference to `google::FlagRegisterer::FlagRegisterer<int>(char const*, char const*, char const*, int*, int*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-vlog_is_on.o):(.text.startup+0x68): undefined reference to `google::FlagRegisterer::FlagRegisterer<int>(char const*, char const*, char const*, int*, int*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-vlog_is_on.o):(.text.startup+0x1c0): undefined reference to `google::FlagRegisterer::FlagRegisterer<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(char const*, char const*, char const*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*)'
/Programs/palo_workspace/baidu/bdg/doris/core/thirdparty/installed/lib64/libglog.a(libglog_la-utilities.o):(.text.startup+0x79): undefined reference to `google::FlagRegisterer::FlagRegisterer<bool>(char const*, char const*, char const*, bool*, bool*)'
```

### What You Expected?

Build BRPC successfully.

### How to Reproduce?

```shell
cd thirdparty
./build-thirdparty.sh --clean
```

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/thirdparty/build-thirdparty.sh
+++ b/thirdparty/build-thirdparty.sh
@@ -805,6 +805,10 @@ build_brpc() {
         ldflags="-L${TP_LIB_DIR} -static-libstdc++ -static-libgcc"
     else
         ldflags="-L${TP_LIB_DIR}"
+
+        # Don't set OPENSSL_ROOT_DIR
+        sed '/set(OPENSSL_ROOT_DIR/,/)/ d' ../CMakeLists.txt >../CMakeLists.txt.bak
+        mv ../CMakeLists.txt.bak ../CMakeLists.txt
     fi
 
     # Currently, BRPC can't be built for static libraries only (without .so). Therefore, we should add `-fPIC`
@@ -812,7 +816,7 @@ build_brpc() {
     LDFLAGS="${ldflags}" \
         "${CMAKE_CMD}" -G "${GENERATOR}" -DBUILD_SHARED_LIBS=1 -DWITH_GLOG=ON -DCMAKE_INSTALL_PREFIX="${TP_INSTALL_DIR}" \
         -DCMAKE_LIBRARY_PATH="${TP_INSTALL_DIR}/lib64" -DCMAKE_INCLUDE_PATH="${TP_INSTALL_DIR}/include" \
-        -DBUILD_BRPC_TOOLS=OFF -DBUILD_SHARED_LIBS=OFF \
+        -DBUILD_BRPC_TOOLS=OFF \
         -DPROTOBUF_PROTOC_EXECUTABLE="${TP_INSTALL_DIR}/bin/protoc" ..
 
     "${BUILD_SYSTEM}" -j "${PARALLEL}"

